### PR TITLE
list + local envoy changes

### PR DESCRIPTION
**Pull request by ilackarms**
BOT NOTES: 
resolves https://github.com/solo-io/wasme/issues/72

### CODE REVIEW

File: pkg/cmd/deploy/deploy.go [hunk 1]
```diff
@@ -158,3 +182,57 @@ func runDeploy(ctx context.Context, opts *options) error {
...
+			return ioutil.NopCloser(bytes.NewBuffer([]byte(local.BasicEnvoyConfig))), nil
+		default:
+			// read file
+			return os.Open(opts.infile)
+		}
+	}()
+	if err != nil {
+		return err
+	}
+	out, err := func() (io.Writer, error) {
```
Comment (yuval-k) [inline pkg/cmd/deploy/deploy.go]: return WriterCloser?
the file created from create needs to be closed

Comment (yuval-k) [reply pkg/cmd/deploy/deploy.go]: other option is to retrurn *os.File as IIRC Stdout is file

File: pkg/deploy/local/local_provider.go [hunk 2]
```diff
@@ -20,98 +26,166 @@ import (
...
+	// additional args passed to the `envoy` command when running Envoy in Docker. Ignored if using DryRyn
+	EnvoyArgs []string
+
+	// the image ref for Envoy to run with docker. Ignored if using DryRyn
+	EnvoyDockerImage string
 }
 
-func (p *Provider) getConfig() (*envoy_config_bootstrap_v2.Bootstrap, error) {
-	b, err := ioutil.ReadAll(p.Input)
+// applies the filter to all selected workloads in selected namespaces
```
Comment (yuval-k) [inline pkg/deploy/local/local_provider.go]: what does namespaces mean in this context?

File: pkg/deploy/local/local_provider.go [hunk 3]
```diff
@@ -20,98 +26,166 @@ import (
...
+	ports, err := getListenerPorts(cfg)
 	if err != nil {
 		return err
 	}
 
-	if err := removeFilterFromListeners(filter, cfg.GetStaticResources().GetListeners()); err != nil {
+	dockerArgs := append([]string{
+		"--rm",
+		"--name", filter.Id,
+		"--entrypoint", "envoy",
```
Comment (yuval-k) [inline pkg/deploy/local/local_provider.go]: i would also add `--disable-hot-restart` here for good measure.

Comment (ilackarms) [reply pkg/deploy/local/local_provider.go]: see `envoyArgs` below

File: pkg/deploy/local/local_provider.go (hunk 2 above)
Comment (ilackarms) [reply pkg/deploy/local/local_provider.go]: fixed comment

**Pull request merged**